### PR TITLE
update roadmap for 2024

### PR DESCRIPTION
**Pull request by THS-on**
This is the initial update for the roadmap. Do we want to include the current Kubernetes efforts?

### CODE REVIEW

**Comment by maugustosilva** [comment]
Q1/Q2 - 2024

Core
* Introduce the concept of "named measured policy", very similar to what already is done to "runtime policies" (IMA). This will require both API and database changes
* Refactoring of `verifier` and `registrar` server API implementation
* Add support for "external actions" during the initial registration
* Completely remove the dependency on `tpm2-tools` from kelyme (one last invocation remains on the `verifier` side)
* Initial implementation of a new `verifier` operating in "push model"

Operator
* Enable a full CI/CD 
* Full support for multi-`verifier` and multi-`registrar`
* Initial implementation of CRD to represent generic (i.e., runtime/IMA, measured boot and even tpm) keylime policies

Q3/Q4
Core
* Add "named tpm policies", completing the transition on having "policies" as a first-class object in keylime, with a life cycle independent of an `agent`
* Continuous refinements and improvements on "push model"
* Harden the "Durable Attestation" implementation, taking the feature out of "experimental" status 

Operator
* Map "named policies" model to CRDs
* Implement `verifier`-driven status updates

**Comment by mheese** [comment]
two things to add to the operator:

- agent scheduler
- agent policy selector

**Comment by THS-on** [comment]
Updated now to include the suggestions by @maugustosilva and @mheese. From my side this is ready to merge

**Review by maugustosilva — APPROVED** [review]

**Comment by niteeshkd** [comment]
Looks good to me.

File: ROADMAP.md
Comment (ruocco) [inline ROADMAP.md:1]: Also change the Title at line 1 with "Keylime Roadmap 2023/2024" ?

Comment (maugustosilva) [reply ROADMAP.md:1]: good catch... @THS-on, can you please change the title? I will merge then.

Comment (THS-on) [reply ROADMAP.md:1]: Thanks I missed that, @maugustosilva is now updated

**Pull request merged**